### PR TITLE
Product - post travel product api

### PR DESCRIPTION
**Pull request by certong0507**

### CODE REVIEW

**Comment by certong0507** [comment]
Hi dudes, i commit the changes for update post travel product api. Please review.

**Comment by jingwong25** [comment]
branch name "ct-post-travel-product-api" please do not put ct infornt.

File: app/controllers/PostTravelProductController.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        
+    }).catch(function(err){
+        return res.render('error', {
+            error: err,
+            status: 500
+        })
+    });
+}
+
+exports.byProfileId = function(req, res, next, id){
```
Comment (jingwong25) [inline app/controllers/PostTravelProductController.js]: `exports.byProfileId = function(req, res, next, id){`

Please put proper API naming.
`exports.byProfileId` 
Change to `exports.getPostTravelProductByProfileId`

File: app/routes/PostTrevalProductRoute.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        .get(postTravelProduct.show);
+
+    app.route('/postTravelProduct/byProfileId/:profileID')
+        .get(postTravelProduct.show);
+
+        
+    app.param('postTravelProductId', postTravelProduct.byId);
+    app.param('productCategoryId', postTravelProduct.byProductCategoryId);
+    app.param('profileID', postTravelProduct.byProfileId);
+};
```
[the remark this answers is not in the document]
Comment (certong0507) [reply app/routes/PostTrevalProductRoute.js]: Updated the naming convention

File: app/controllers/PostTravelProductController.js (hunk 1 above)
Comment (certong0507) [reply app/controllers/PostTravelProductController.js]: Updated the naming convention

File: app/models/PostTravelProductModels.js [hunk 3]
```diff
@@ -18,7 +18,8 @@ module.exports = function (sequelize, DataTypes) {
         },
         amount: {
             type: DataTypes.DECIMAL(10, 3),
-            allowNull: false
+            allowNull: false,
+            field: 'amount'
```
[the remark this answers is not in the document]
Comment (certong0507) [reply app/models/PostTravelProductModels.js]: Removed additional code

File: app/routes/PostTravelProductRoute.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const postTravelProduct = require('../controllers/PostTravelProductController');
+module.exports = function (app) {
+
+    app.route('/postTravelProduct/byId/:postTravelProductId')
```
Comment (jingwong25) [inline app/routes/PostTravelProductRoute.js]: app.route('/postTravelProduct/id/:postTravelProductId')

File: app/routes/PostTravelProductRoute.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        .get(postTravelProduct.show)
+        .put(postTravelProduct.updateProduct);
+
+    app.route('/postTravelProduct')
```
Comment (jingwong25) [inline app/routes/PostTravelProductRoute.js]: app.route('/posttravelproduct')

File: app/routes/PostTravelProductRoute.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+        .get(postTravelProduct.all)
+        .post(postTravelProduct.createProduct, postTravelProduct.createProductDocument);
+
+    app.route('/postTravelProduct/byProductCategoryId/:productCategoryId')
```
Comment (jingwong25) [inline app/routes/PostTravelProductRoute.js]: app.route('/posttravelproduct/productcategory/:productCategoryId')

File: app/routes/PostTravelProductRoute.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+        .get(postTravelProduct.show);
+
+    app.route('/postTravelProduct/:userProfileId')
```
Comment (jingwong25) [inline app/routes/PostTravelProductRoute.js]: app.route('/posttravelproduct/:profileId')

Comment (jingwong25) [reply app/routes/PostTravelProductRoute.js]: No need to change to userProfileId as profileId can be use.

Comment (certong0507) [reply app/routes/PostTravelProductRoute.js]: Cannot have the same api name in the project

File: app/routes/PostTravelProductRoute.js (hunk 6 above)
Comment (certong0507) [reply app/routes/PostTravelProductRoute.js]: Updated already

File: app/routes/PostTravelProductRoute.js (hunk 5 above)
Comment (certong0507) [reply app/routes/PostTravelProductRoute.js]: Updated already

File: app/routes/PostTravelProductRoute.js (hunk 4 above)
Comment (certong0507) [reply app/routes/PostTravelProductRoute.js]: Updated already

File: app/routes/PostTravelProductRoute.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const postTravelProduct = require('../controllers/PostTravelProductController');
+module.exports = function (app) {
+
+    app.route('/posttravelproduct/byid/:postTravelProductId')
```
Comment (jingwong25) [inline app/routes/PostTravelProductRoute.js]: required to standardize the url naming

File: app/routes/PostTravelProductRoute.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+    app.route('/posttravelproduct/byproductcategoryid/:productCategoryId')
+        .get(postTravelProduct.show);
+
+    app.route('/posttravelproduct/:userProfileId')
+        .get(postTravelProduct.show);
+
+    app.param('postTravelProductId', postTravelProduct.getPostTravelProductById);
+    app.param('productCategoryId', postTravelProduct.getPostTravelProductByProductCategoryId);
+    app.param('userProfileId', postTravelProduct.getPostTravelByUserProfileId);
```
Comment (jingwong25) [inline app/routes/PostTravelProductRoute.js]: `app.param('userProfileId', postTravelProduct.getPostTravelByUserProfileId);`
never ending changing parameter naming.

**Pull request merged**